crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -290,32 +290,35 @@ where
         let object_history_changes: TransactionObjectChangesToCommit =
             Self::index_objects_history(data.clone(), package_resolver.clone()).await?;
 
-        let (checkpoint, db_transactions, db_events, db_indices, db_displays) = {
+        let (checkpoint, db_transactions, db_events, db_indices, db_displays, cp_tx) = {
             let CheckpointData {
                 transactions,
                 checkpoint_summary,
                 checkpoint_contents,
             } = data;
 
-            let (db_transactions, db_events, db_indices, db_displays) = Self::index_transactions(
-                transactions,
-                &checkpoint_summary,
-                &checkpoint_contents,
-                &metrics,
-            )
-            .await?;
+            let (db_transactions, db_events, db_indices, db_displays, cp_tx) =
+                Self::index_transactions(
+                    transactions,
+                    &checkpoint_summary,
+                    &checkpoint_contents,
+                    &metrics,
+                )
+                .await?;
 
             let successful_tx_num: u64 = db_transactions.iter().map(|t| t.successful_tx_num).sum();
             (
                 IndexedCheckpoint::from_sui_checkpoint(
                     &checkpoint_summary,
                     &checkpoint_contents,
                     successful_tx_num as usize,
+                    cp_tx,
```

Comment:
I think this is the only place where a data structure like `CpToTx` might be helpful, but if that's the case, simply passing the two sequence numbers into `from_sui_checkpoint` probably suffices.